### PR TITLE
OPENTOK-46938 add blur options to publisher filters

### PR DESCRIPTION
**Pull request by maikthomas**
#### What is this PR doing?
Add background blur with low and high blur strength as options in the publisher filter picker.

#### How should this be manually tested?
- Run locally
- Check that filters work and that all transitions between filters work i.e.
- between blur and none
- between high and low blur
- between blur and existing color filters
- between none and color filters
- blur effects should be cleared when selected other filters (the clear method should be called, you can't tell just by looking at the publisher because the current filters actually create a new publisher, so the blur effect wouldn't be visible anyway).

#### What are the relevant tickets?
https://jira.vonage.com/browse/OPENTOK-46938

### CODE REVIEW

File: src/js/controllers.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ angular.module('opentok-meet').controller('RoomCtrl', ['$scope', '$http', '$wind
       fixedRatio: !$scope.zoomed,
     };
     $scope.filter = 'none';
+    $scope.filterousFilters = ['brannan', 'xpro2', 'moon', 'nashville', 'lofi', 'helena', '1977'];
```
Comment (maikthomas) [inline src/js/controllers.js:31]: I added this to the scope for the show/hide 'filtered-publisher' element logic

File: src/js/filter-picker.js [hunk 2]
```diff
@@ -6,7 +6,9 @@ function FilterPickerDirective() {
   function link(scope, element) {
     let currentPublisher;
     scope.showFilterList = false;
-    scope.filters = ['none', 'brannan', 'xpro2', 'moon', 'nashville', 'lofi', 'helena', '1977'];
+    const nativePublisherFilters = ['BG Blur: Low', 'BG Blur: High'];
+    scope.filterousFilters = ['brannan', 'xpro2', 'moon', 'nashville', 'lofi', 'helena', '1977'];
```
Comment (maikthomas) [inline src/js/filter-picker.js:10]: I don't know enough about angular to know why it isn't enough just to have this here and not in controllers.js

File: src/js/filter-picker.js [hunk 3]
```diff
@@ -70,6 +89,16 @@ function FilterPickerDirective() {
...
+
+
+      // TODO: Remove these mocks once OT API is implemented
+      currentPublisher.applyVideoFilter = (...args) => {
+        console.log('applyVideoFilter called with ', ...args);
+      };
+      currentPublisher.clearVideoFilter = () => {
+        console.log('clear called');
+      };
+      // END TODO
```
Comment (maikthomas) [inline src/js/filter-picker.js:101]: Until we have the API this is useful to debug what's going on

Comment (v-kpheng) [reply src/js/filter-picker.js:101]: What if we call the APIs and just wrap them in `try`/`catch` blocks?  That way, once the API is published, it'll just work.  WDYT?

**Comment by v-kpheng** [comment]
@maikthomas, is this ready for review again?

**Comment by v-kpheng** [comment]
@maikthomas, any update?  Please advise.  Thanks! 🙏

**Comment by maikthomas** [comment]
@v-kpheng yep I just tested this with your latest blur changes and it looks like it's working well 👍 
Please re-review

File: src/js/controllers.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+const filterousFilters = require('./filterous-filters');
```
Comment (v-kpheng) [inline src/js/controllers.js:1]: `filterousFilters`?  Seems redundant, IMO.

**Review by v-kpheng — APPROVED** [review]
LGTM! 💪 🚀

File: src/js/controllers.js (hunk 4 above)
Comment (maikthomas) [reply src/js/controllers.js:1]: as opposed to all filters, specifically not the native ones

**Comment by v-kpheng** [comment]
Tested this locally.  Looks good! 💪 
Awesome work, Mike! 💪 

image

**Pull request merged**